components/app_bar/app_bar.tsx

```diff
@@ -24,26 +25,32 @@ export default function AppBar() {
         return null;
     }
 
-    // Order the plugin components so that Playbooks is the first one,
-    // Boards is the second one, and the others remain in the same place.
-    const orderedAppBarPluginComponents = [...appBarPluginComponents];
-    orderedAppBarPluginComponents.sort((a, b) => {
-        switch (a.pluginId) {
-        case 'playbooks':
-            return -1;
+    const coreProductsIds = ['playbooks', 'focalboard'];
 
-        case 'focalboard':
-            return b.pluginId === 'playbooks' ? 1 : -1;
+    // The type guard in the filter (which removes all undefined elements) is needed for
+    // Typescript to correctly type coreProducts.
+    const coreProducts = coreProductsIds.
+        map((id) => appBarPluginComponents.find((element) => element.pluginId === id)).
+        filter((element): element is PluginComponent => Boolean(element));
 
-        default:
-            return 1;
-        }
-    });
+    const appBarPluginComponentsWithoutCoreProducts = appBarPluginComponents.filter((element) => !coreProductsIds.includes(element.pluginId));
+    const pluginComponents = appBarPluginComponentsWithoutCoreProducts.concat(channelHeaderComponents);
 
-    const pluginComponents = orderedAppBarPluginComponents.concat(channelHeaderComponents);
+    const isCoreProductsSectionNonEmpty = coreProducts.length > 0;
+    const isRegularIconsSectionNonEmpty = pluginComponents.length > 0 || appBarBindings.length > 0;
+    const isDividerVisible = isCoreProductsSectionNonEmpty && isRegularIconsSectionNonEmpty;
 
     return (
         <div className={'app-bar'}>
+            {coreProducts.map((product) => (
+                <AppBarPluginComponent
+                    key={product.id}
+                    component={product}
+                />
+            ))}
+            {isDividerVisible &&
+            <hr className={'app-bar__divider'}/>
+            }
```

Comment:
Just my own preference 0/5

```suggestion
            {isDividerVisible && (
                <hr className={'app-bar__divider'}/>
            )}
```